src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2462,6 +2463,7 @@ CoPartitionedTables(Oid firstRelationId, Oid secondRelationId)
 	uint32 firstListShardCount = firstTableCache->shardIntervalArrayLength;
 	uint32 secondListShardCount = secondTableCache->shardIntervalArrayLength;
 	FmgrInfo *comparisonFunction = firstTableCache->shardIntervalCompareFunction;
+	Oid collation = firstTableCache->partitionColumn->varcollid;
```

Comment:
What if the tables don't have the same collation Id? I think we should return false in that case since we really don't know their ordering match? Maybe also add a test for this?

Other option is to handle this check in call sites of this function and add an assertion here.

Reply:
I figured it was correct since we assume `comparisonFunction` is compatible between the table caches. It doesn't appear so, so I've added a commit that checks that both columns have the same type oid & collation oid